satpy/etc/readers/viirs_sdr.yaml

```diff
@@ -14,6 +14,7 @@ datasets:
     file_key: All_Data/{dataset_group}_All/Longitude
     units: "degrees"
     standard_name: longitude
+    coordinates: [i_longitude, i_latitude]
```

Comment:
Isn't this creating an infinite recursion ?

Reply:
Based on the way you implemented the stuff in the base reader, no. All the coordinates are collected for all datasets, then they are assigned.